compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs

```diff
@@ -416,12 +416,34 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                         _,
                     ) = pat.kind
                 {
-                    err.span_suggestion(
-                        upvar_ident.span,
-                        "consider changing this to be mutable",
-                        format!("mut {}", upvar_ident.name),
-                        Applicability::MachineApplicable,
-                    );
+                    if upvar_ident.name.as_str() == "self" {
```

Comment:
Sorry about this, I don't know how to check if an upvar corresponds to an implicit self, so this condition is just a very bad attempt at it.

Reply:
This is fine probably